client/extensions/woocommerce/state/data-layer/ui/shipping-zones/index.js

```diff
@@ -143,10 +144,15 @@ const getAutoOrderZonesSteps = ( siteId, zoneId, serverZones, allServerLocations
 	} ).filter( Boolean ) );
 };
 
-const getZoneMethodDeleteSteps = ( siteId, zoneId, method ) => {
+const getZoneMethodDeleteSteps = ( siteId, zoneId, method, state ) => {
+	const methodType = getShippingZoneMethod( state, method.id, siteId ).methodType;
+
 	return [ {
 		description: translate( 'Deleting Shipping Method' ),
 		onStep: ( dispatch, actionList ) => {
+			analytics.tracks.recordEvent( 'calypso_woocommerce_shipping_method_deleted', {
+				shippingMethod: methodType,
```

Comment:
property names need to be lowercase and separated by underscores. While testing this, I noticed that the events are going into the "rejects" table -- and I think the property naming is the reason for that.

https://fieldguide.automattic.com/tracks-naming-conventions/